### PR TITLE
FLEX-154: Added query tests and fixed a few query bugs

### PR DESCRIPTION
**Pull request by mjsalinger**

### CODE REVIEW

File: test/lib/modules/query.test.js [hunk 1]
```diff
@@ -12,9 +12,1491 @@
...
+    });
+
+    it('should return the query', () => {
+      const query = new Query().mod(randomString(), 5);
+      query.should.be.an.instanceOf(Query);
+    });
+  });
+
+  describe('matches()', () => {
+    it('throw an error for unsupported ignoreCase option', () => {
```
Comment (samlevin) [inline test/lib/modules/query.test.js]: Does this need a test for `null` case?
